### ISSUE TITLE
hyper parameter dictionary doesn't support another dictionary as a value

### ISSUE THREAD

**Issue by Idodox** [comet-ml/issue-tracking#349]
Not sure if this is a bug or feature request.

One of my hyper parameter dictionary values is the parameters for learning rate scheduler, as follows:

"lr_scheduler": {'step_size': 10, 'gamma': 0.5}

in the comet interface this is only saved as: ["step_size","gamma"]

**Comment by dsblank** [comment]
Yeah, of all the things we could do, that doesn't seem useful. Thanks for reporting! It is now on out list.

**Comment by dsblank** [comment]
As a work around for the moment, two ideas:

1. Log the dict as a string:

```python
experiment.log_parameter("lr_scheduler", str(lr_scheduler))
```

which would log "lr_scheduler" as "{'step_size': 10, 'gamma': 0.5}"

2. Log as a nested parameters:

```python
def log_nested_dictionary(experiment, nested_dictionary, prefix=""):
    for key in nested_dictionary.keys():
        if isinstance(nested_dictionary[key], dict):
            log_nested_dictionary(experiment, nested_dictionary[key], prefix + "." + key)
        else:
            experiment.log_parameter(prefix + "." + key, nested_dictionary[key])
```

and then use:

```python
log_nested_dictionary(experiment, lr_dict, "lr_scheduler")
```

which would log "lr_scheduler.step_size" as 10, and "lr_scheduler.step_gamma" as 0.5.

**Comment by dsblank** [comment]
@Idodox This should be fixed in the latest comet_ml release, version 3.2.0. Thanks for reporting!
